senior_game_client/src/http.rs

```diff
@@ -0,0 +1,167 @@
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use bevy::prelude::*;
+#[cfg(test)]
+use mockito;
+use reqwest::blocking::{Client, RequestBuilder, Response};
+use reqwest::{Error, StatusCode};
+use serde_json::{Map, Value};
+use url::Url;
+
+#[allow(dead_code)]
+pub enum WebRequestVerb {
+  Get,
+  Post,
+}
+
+pub const LOGIN_SUFFIX: &str = "/session";
+
+pub fn domain_url() -> Url {
+  #[cfg(not(test))]
+  let domain: &str = "https://accounts.senior.realliance.net";
+
+  #[cfg(test)]
+  let domain: &str = &mockito::server_url();
+
+  Url::parse(domain).unwrap()
+}
+
+pub fn login_route() -> String {
+  return domain_url()
+    .join(LOGIN_SUFFIX)
+    .unwrap()
+    .as_str()
+    .to_string();
```

Comment:
is as_str() needed here if .to_string() is already used?

Reply:
`as_str` converts from a url object to a `&str`, which was then converted into a `String`. However, upon further reading into the docs. `into_string` can be used instead